redgenes/gene_annotations.py

```diff
@@ -77,12 +81,15 @@ def run_barrnap(input_fasta, outdir, cpu=1):
     return str(barrnap_output_gff), commands_str
 
 
-def extract_md_info(tsv_file):
-    """Load data from tsv and checks data integrity."""
-    df = pd.read_csv(tsv_file, sep="\t", dtype=str)
+def extract_md_info(md_str):
+    """Load data from input and checks data integrity."""
+    data = md_str.split("\t")
+    data_with_nan = [np.nan if x == "" else x for x in data]
+    df = pd.DataFrame([data_with_nan], columns=MD_HEADER)
+    # df = pd.read_csv(tsv_file, sep="\t", dtype=str)
```

Comment:
Wouldn't empty strings show up as such in the dataframe already, or as `None`?

Reply:
Yea that's what I assumed but test runs showed that empty strings were stored as '' instead of None and errors out in the subsequent `df = df.astype(dtype_map)` with an error `ValueError: invalid literal for int() with base 10: ''`.

E.g. data (tab splitted)
`['hA11.fna', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '(Gordon) Dey N, Wagner VE, Blanton LV, Cheng J, Fontana L, Haque R, Ahmed T, Gordon JI. Regulators of gut motility revealed by a gnotobiotic model of diet-microbiome interactions related to travel. Cell. 2015 Sep 24;163(1):95-107. doi: 10.1016/j.cell.2015.08.059. Erratum in: Cell. 2015 Nov 5;163(4):1037. PMID: 26406373; PMCID: PMC4583712.', '', '', '/projects/gordon_genome/Dey_et_al_genomes/ENA_contigs/hA11.fna.gz', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', 'Gordon', '/projects/greengenes2/gg2_genomes/external/hA11.fna.gz']`

df if using `df = pd.DataFrame([data], columns=MD_HEADER)`
  assembly_accession bioproject biosample wgs_master refseq_category taxid  ... total_gene_count protein_coding_gene_count non_coding_gene_count pubmed_id  source                                         local_path
0           hA11.fna                                                        ...                                                                             Gordon  /projects/greengenes2/gg2_genomes/external/hA1...

[1 rows x 40 columns]